test/e2e/run-openrpc-api-test-coverage.ts

```diff
@@ -0,0 +1,418 @@
+        //   description: 'Example of a pending getTransactionCount request',
+        //   params: [
+        //     {
+        //       name: 'address',
+        //       value: ACCOUNT_1,
+        //     },
+        //     {
+        //       name: 'tag',
+        //       value: 'pending',
+        //     },
+        //   ],
+        //   result: {
+        //     name: 'getTransactionCountResult',
+        //     value: '0x0',
+        //   },
+        // },
+      ];
+
+      const server = mockServer(port, openrpcDocument);
+      server.start();
+
+      // TODO: move these to a "Confirmation" tag in api-specs
+      const methodsWithConfirmations = [
+        'wallet_requestPermissions',
+        'eth_requestAccounts',
+        'wallet_watchAsset',
+        'personal_sign', // requires permissions for eth_accounts
+        'wallet_addEthereumChain',
+        'eth_signTypedData_v4', // requires permissions for eth_accounts
+        'wallet_switchEthereumChain',
+
+        // commented out because its not returning 4001 error.
+        // see here https://github.com/MetaMask/metamask-extension/issues/24227
+        // 'eth_getEncryptionPublicKey', // requires permissions for eth_accounts
+      ];
+      class VerboseConsoleReporter extends ConsoleStreamingReporter {
+        onTestBegin(_: unknown, call: Call) {
+          console.log('onTestBegin', call.methodName, call.params);
+        }
+      }
+
+      const filteredMethods = openrpcDocument.methods
+        .filter((_m: unknown) => {
+          const m = _m as MethodObject;
+          return (
+            m.name.includes('snap') ||
+            m.name.includes('Snap') ||
+            m.name.toLowerCase().includes('account') ||
+            m.name.includes('crypt') ||
+            m.name.includes('blob') ||
+            m.name.includes('sendTransaction') ||
+            m.name.startsWith('wallet_scanQRCode') ||
+            methodsWithConfirmations.includes(m.name) ||
+            // filters are currently 0 prefixed for odd length on
+            // extension which doesn't pass spec
+            // see here: https://github.com/MetaMask/eth-json-rpc-filters/issues/152
+            m.name.includes('filter') ||
+            m.name.includes('Filter')
+          );
+        })
+        .map((m) => (m as MethodObject).name);
+
+      const testCoverageResults = await testCoverage({
+        openrpcDocument: (await parseOpenRPCDocument(
+          openrpcDocument as never,
+        )) as never,
+        transport,
+        reporters: [
+          new VerboseConsoleReporter(),
+          new HtmlReporter({ autoOpen: !process.env.CI }),
+        ],
+        skip: [
+          'eth_coinbase',
+          // these 2 methods below are not supported by MetaMask extension yet and
+          // don't get passed through. See here: https://github.com/MetaMask/metamask-extension/issues/24225
+          'eth_getBlockReceipts',
+          'eth_maxPriorityFeePerGas',
```

Comment:
kindof weird this isn't passed through and supported...? 🤷

Reply:
we only pass through a whitelist of methods

Reply:
ideally this whitelist should be grabbed from the API specs